clinica/utils/image.py

```diff
@@ -0,0 +1,135 @@
+from os import PathLike
+from pathlib import Path
+from typing import Callable, Optional, Tuple, Union
+
+import nibabel as nib
+import numpy as np
+from nibabel.nifti1 import Nifti1Image
+
+
+def compute_aggregated_volume(
+    image_filename: PathLike,
+    aggregator: Optional[Callable] = None,
+    volumes_to_keep: Optional[np.ndarray] = None,
+) -> np.ndarray:
+    """Computes the aggregated 3D volumes from a 4D image and an aggregator function.
+
+    The aggregation is computed on the last (fourth) dimension.
+    It is possible to compute the aggregation on a subset of the volumes through the
+    parameter 'volumes_to_keep'.
+
+    Parameters
+    ----------
+    image_filename : str
+        The path to the input image.
+
+    aggregator : Callable, optional
+        The aggregator function. Example: np.average, np.median.
+        If None, no aggregation will be performed.
+
+    volumes_to_keep : np.array, optional
+        The volumes to be kept during aggregation. This is a 1D index array.
+        If None, all volumes of the input image will be kept during aggregation.
+        Default = None.
+
+    Returns
+    -------
+    np.ndarray:
+        The 3D volume data array obtained from the aggregation.
+    """
+    volumes = np.array(nib.four_to_three(nib.load(image_filename)))
+    if volumes_to_keep is None:
+        volumes_to_keep = volumes
+    else:
+        volumes_to_keep = volumes[volumes_to_keep]
+    data = [
+        volume.get_fdata(dtype="float32").astype(np.float32)
+        for volume in volumes_to_keep
+    ]
+    if aggregator is None:
+        return np.stack(data, axis=-1)
+    return aggregator(data, axis=0)
+
+
+def get_new_image_like(old_image: PathLike, new_image_data: np.ndarray) -> Nifti1Image:
```

Comment:
I have trouble understanding this function's name. Am I meant to read it with the first argument ?

Reply:
The naming logic comes from the `new_img_like` function of Nilearn to which I got accustomed over the years. It basically means: "Out of the provided data array, create a new nifti image with the same affine and header as the provided image". Since it is part of the public API of Nilearn, the name had to be relatively short, but we don't have this problem here.
So, if it doesn't feel clear, we can change it for something more informative like `build_new_image_from_data_array_using_old_image` or `create_image_from_existing_image_and_new_data`....
I'm open to suggestions ;)

Reply:
I guess it follows the NumPy naming convention (`zero_like`, `empty_like`) which constructs an instance from the shape (or metadata) of another instance? In that case, that's a good tradeoff between expressive and short naming.

Reply:
I'm pretty sure you're right about this since Gael was probably the one writing this function a long time ago. I don't have a strong preference as long as the name is meaningful for users and contributors.